src/webgpu/util/device_pool.ts

```diff
@@ -129,6 +129,20 @@ export class DevicePool {
           await attemptGarbageCollection();
         }
       }
+
+      // In the try block, we may throw an error if the device is lost in order to force device
+      // reinitialization, however, if the device lost was expected we want to suppress the error
+      // The device lost is expected when `holder.expectedLostReason` is equal to
+      // `holder.lostInfo.reason`.
+      const expectedDeviceLost =
+        holder.expectedLostReason !== undefined &&
+        holder.lostInfo !== undefined &&
+        holder.expectedLostReason === holder.lostInfo.reason;
+      if (!expectedDeviceLost) {
+        // Throws unexpected device lost, as well as unexpected validation/out-of-memory/internal errors
+        // from popErrorScope.
+        throw ex;
+      }
```

Comment:
uh. whoops, my bad, I clearly did not test this well enough. I wanted to move the expectedDeviceLost logic out of the `catch`. Would it work to do simply:

```suggestion
      throw ex;
```